MBBSEmu/HostProcess/HostRoutines/FsdRoutines.cs

```diff
@@ -596,7 +637,27 @@ private void InFullScreenDisplay(SessionBase session)
                             //Hitting Enter on the last Field
                             if (_fsdFields[session.Channel].SelectedOrdinal ==
                                 _fsdFields[session.Channel].Fields.Count - 1)
-                                return;
+                            {
+                                //Final Field in FSE is for Savings
+                                if (session.SessionState == EnumSessionState.InFullScreenEditor)
+                                {
+                                    var fseFinalState =
+                                        session.CurrentModule.Memory.GetOrAllocateVariablePointer(
+                                            $"FSE-FinalStateCode-{session.Channel}", sizeof(ushort));
+                                    switch (_fsdFields[session.Channel].SelectedField.Value)
+                                    {
+                                        case "SAVE":
+                                            session.CurrentModule.Memory.SetWord(fseFinalState, 0);
```

Comment:
can fseFinalState be null?

Reply:
The pointer? No, it's declared prior to the switch.

The logic here is basically saying, "if you're on the last field of the FSE FSD template AND hit enter, evaluate the input" as it's always the QUIT/SAVE multi-select. The valid return values according to the docs are `0` and `-1`